cdap-master/src/main/java/co/cask/cdap/data/runtime/main/MasterTwillApplication.java

```diff
@@ -401,4 +408,35 @@ public boolean apply(File file) {
     // Depending on how the cluster is configured, we might need to localize the MR framework tgz as well.
     extraClassPath.addAll(MapReduceContainerHelper.localizeFramework(hConf, localizeResources));
   }
+
+  /**
+   * Prepares the {@link HBaseDDLExecutor} implementation for localization.
+   */
+  private void prepareHBaseDDLExecutorResources(Path tempDir, CConfiguration cConf,
+                                                Map<String, LocalizeResource> localizeResources) throws IOException {
+    String ddlExecutorExtensionDir = cConf.get(Constants.HBaseDDLExecutor.EXTENSIONS_DIR);
+    if (ddlExecutorExtensionDir == null) {
+      // Nothing to localize
+      return;
+    }
+
+    List<File> files = DirUtils.listFiles(new File(ddlExecutorExtensionDir));
+    // Currently we only support single HBase DDL extension
+    File moduleDir = files.get(0);
```

Comment:
We don't need this check, right? Since `ddlExecutorExtensionDir` is already pointing to `ext/hbase`, all you need is just to create a `jar` file of that directory. Then you can simplify the logic to:

```java
BundleJarUtil.createJar(new File(ddlExecutorExtDir), target);
localizeResources.put(target.getName(), new LocalizeResource(target, true));
cConf.set(Constants.HBaseDDLExecutor.EXTENSIONS_DIR, target.getName());
```